### ISSUE TITLE
VTT subtitles for viafree.se not downloaded correctly for some episodes

### ISSUE THREAD

**Issue by filigranen** [spaam/svtplay-dl#1332]
I'm having trouble downloading subtitles for some viafree episodes (actually, this one was the first I tried, so not sure how big the problem is).

@Sopor  in [issue #553 ](https://github.com/spaam/svtplay-dl/issues/553) has the same problem, m3u8 link does not get resolved to the real vtt, and it does not seem to be fixed yet.
Perhaps I should just comment that issue? Not sure if it gets seen if I do, since it's buried four years back?

Like @Sopor I get the same behavior, on linux.
Haven't tried it on windows, but that shouldn't matter like you say in the original issue.

```
[tomteverkstan] ~/svtviafreesub
(09:35) (svtdl:0) $ svtplay-dl --version
svtplay-dl 3.0

[tomteverkstan] ~/svtviafreesub
(09:36) (svtdl:0) $ uname -a
Linux tomteverkstan 5.11.1-arch1-1 #1 SMP PREEMPT Tue, 23 Feb 2021 14:05:30 +0000 x86_64 GNU/Linux

[tomteverkstan] ~/svtviafreesub
(09:39) (svtdl:0) $ python --version
Python 3.9.2

[tomteverkstan] ~/svtviafreesub
(09:39) (svtdl:0) $ cat ~/.svtplay-dl.yaml
default:
  filename: "{title}.s{season}e{episode}.{episodename}-{id}-{service}.{ext}"
  exclude: teckentolk,syntolk
service:
    viaplay:
        username: my@user.info
        password: mypassword
```

**This one works**, m3u8 gets translated to vtt correctly (same link as the last post in #533:
```
[tomteverkstan] ~/svtviafreesub
(09:41) (svtdl:0) $ svtplay-dl --verbose -S --force-subtitle https://www.viafree.se/program/reality/geordie-shore/sasong-17/avsnitt-3
DEBUG [1614847366.929518] /usr/lib/python3.9/site-packages/svtplay_dl/utils/getmedia.py/get_media: version: 3.0
DEBUG [1614847366.9317875] /usr/lib/python3.9/site-packages/svtplay_dl/service/__init__.py/__init__: service: viaplay
DEBUG [1614847366.9318275] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://www.viafree.se/program/reality/geordie-shore/sasong-17/avsnitt-3'
DEBUG [1614847366.9328778] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): www.viafree.se:443
DEBUG [1614847368.3836203] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://www.viafree.se:443 "GET /program/reality/geordie-shore/sasong-17/avsnitt-3 HTTP/1.1" 200 None
DEBUG [1614847368.3870835] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://viafree.mtg-api.com/identity/viafree/auth/pwd/sessions'
DEBUG [1614847368.3882253] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): viafree.mtg-api.com:443
DEBUG [1614847368.6982665] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://viafree.mtg-api.com:443 "POST /identity/viafree/auth/pwd/sessions HTTP/1.1" 200 None
DEBUG [1614847368.7021604] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://viafree.mtg-api.com/stream-links/viafree/web/se/clear-media-guids/20516216/streams'
DEBUG [1614847368.8934672] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://viafree.mtg-api.com:443 "GET /stream-links/viafree/web/se/clear-media-guids/20516216/streams HTTP/1.1" 200 798
DEBUG [1614847368.8941925] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://vod7-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net/11f08a1/20516216-1532570214139.ism/master.m3u8?start_index=1&filter=%28type%3D%3D%22audio%22%26%26%28FourCC%3D%3D%22AACL%22%29%29%7C%7C%28type%3D%3D%22video%22%26%26FourCC%3D%3D%22AVC1%22%29%7C%7C%28type%3D%3D%22textstream%22%26%26%28systemLanguage%3D%3D%22dan%22%29%7C%7C%28systemLanguage%3D%3D%22fin%22%29%7C%7C%28systemLanguage%3D%3D%22nor%22%29%7C%7C%28systemLanguage%3D%3D%22swe%22%29%29&hdnts=st=1614847308~exp=1614933768~acl=*~hmac=af7fab5c0c62ad9aebc27040842c46c332e1addc307611e8e3cfa908f1c9ac40'
DEBUG [1614847368.897644] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): vod7-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net:443
DEBUG [1614847369.0146127] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://vod7-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net:443 "GET /11f08a1/20516216-1532570214139.ism/master.m3u8?start_index=1&filter=%28type%3D%3D%22audio%22%26%26%28FourCC%3D%3D%22AACL%22%29%29%7C%7C%28type%3D%3D%22video%22%26%26FourCC%3D%3D%22AVC1%22%29%7C%7C%28type%3D%3D%22textstream%22%26%26%28systemLanguage%3D%3D%22dan%22%29%7C%7C%28systemLanguage%3D%3D%22fin%22%29%7C%7C%28systemLanguage%3D%3D%22nor%22%29%7C%7C%28systemLanguage%3D%3D%22swe%22%29%29&hdnts=st=1614847308~exp=1614933768~acl=*~hmac=af7fab5c0c62ad9aebc27040842c46c332e1addc307611e8e3cfa908f1c9ac40 HTTP/1.1" 200 None
DEBUG [1614847369.0206547] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://cdn-subtitles-mtgx-tv.akamaized.net/pitcher/20xxxxxx/2051xxxx/205162xx/20516216/0500131436/20516216_sv.vtt'
DEBUG [1614847369.0226092] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): cdn-subtitles-mtgx-tv.akamaized.net:443
DEBUG [1614847369.1418812] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://cdn-subtitles-mtgx-tv.akamaized.net:443 "GET /pitcher/20xxxxxx/2051xxxx/205162xx/20516216/0500131436/20516216_sv.vtt HTTP/1.1" 200 47195
INFO [1614847369.172759] /usr/lib/python3.9/site-packages/svtplay_dl/utils/output.py/output: Outfile: geordie.shore.s17e03.avsnitt.3-958013-viaplay.srt

[tomteverkstan] ~/svtviafreesub
(09:42) (svtdl:0) $ head -n 4 geordie.shore.s17e03.avsnitt.3-958013-viaplay.srt 
1
00:00:00,000 --> 00:00:04,120
Programmet innehåller grovt språk,
våldsamt beteende–
```

This one **DOES NOT** work:
```
[tomteverkstan] ~/svtviafreesub
(09:42) (svtdl:0) $ svtplay-dl --verbose -S --force-subtitle https://www.viafree.se/program/drama/morden-i-midsomer/sasong-16/avsnitt-3
DEBUG [1614847759.5128746] /usr/lib/python3.9/site-packages/svtplay_dl/utils/getmedia.py/get_media: version: 3.0
DEBUG [1614847759.5150814] /usr/lib/python3.9/site-packages/svtplay_dl/service/__init__.py/__init__: service: viaplay
DEBUG [1614847759.5151215] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://www.viafree.se/program/drama/morden-i-midsomer/sasong-16/avsnitt-3'
DEBUG [1614847759.5161662] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): www.viafree.se:443
DEBUG [1614847759.913993] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://www.viafree.se:443 "GET /program/drama/morden-i-midsomer/sasong-16/avsnitt-3 HTTP/1.1" 200 None
DEBUG [1614847759.9175372] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://viafree.mtg-api.com/identity/viafree/auth/pwd/sessions'
DEBUG [1614847759.918831] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): viafree.mtg-api.com:443
DEBUG [1614847760.2095394] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://viafree.mtg-api.com:443 "POST /identity/viafree/auth/pwd/sessions HTTP/1.1" 200 None
DEBUG [1614847760.2134697] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://viafree.mtg-api.com/stream-links/viafree/web/se/clear-media-guids/V09140/streams'
DEBUG [1614847760.3603702] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://viafree.mtg-api.com:443 "GET /stream-links/viafree/web/se/clear-media-guids/V09140/streams HTTP/1.1" 200 983
DEBUG [1614847760.3610985] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://vod9-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net/e6ca938/V09140-1611600129762.ism/master.m3u8?start_index=1&filter=%28type%3D%3D%22audio%22%26%26%28FourCC%3D%3D%22AACL%22%29%29%7C%7C%28type%3D%3D%22video%22%26%26FourCC%3D%3D%22AVC1%22%29%7C%7C%28type%3D%3D%22textstream%22%26%26%28systemLanguage%3D%3D%22swe%22%29%29&hdnts=st=1614847700~exp=1614934160~acl=*~hmac=457347f680345aa15acff1ef5167ea725e83faa2d9806d047536c8bfd4af5d56'
DEBUG [1614847760.3640645] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): vod9-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net:443
DEBUG [1614847760.4844954] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://vod9-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net:443 "GET /e6ca938/V09140-1611600129762.ism/master.m3u8?start_index=1&filter=%28type%3D%3D%22audio%22%26%26%28FourCC%3D%3D%22AACL%22%29%29%7C%7C%28type%3D%3D%22video%22%26%26FourCC%3D%3D%22AVC1%22%29%7C%7C%28type%3D%3D%22textstream%22%26%26%28systemLanguage%3D%3D%22swe%22%29%29&hdnts=st=1614847700~exp=1614934160~acl=*~hmac=457347f680345aa15acff1ef5167ea725e83faa2d9806d047536c8bfd4af5d56 HTTP/1.1" 200 None
DEBUG [1614847760.491114] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://cdn-subtitles-mtgx-tv.akamaized.net/pitcher/V0xxxx/V091xx/V09140/6070025811/V09140_sub_sv.m3u8'
DEBUG [1614847760.4920683] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): cdn-subtitles-mtgx-tv.akamaized.net:443
DEBUG [1614847760.6047904] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://cdn-subtitles-mtgx-tv.akamaized.net:443 "GET /pitcher/V0xxxx/V091xx/V09140/6070025811/V09140_sub_sv.m3u8 HTTP/1.1" 200 1249
INFO [1614847760.607034] /usr/lib/python3.9/site-packages/svtplay_dl/utils/output.py/output: Outfile: morden.i.midsomer.s16e03.avsnitt.3-685979-viaplay.srt

[tomteverkstan] ~/svtviafreesub
(09:49) (svtdl:0) $ cat morden.i.midsomer.s16e03.avsnitt.3-685979-viaplay.srt 
#EXTM3U
#EXT-X-TARGETDURATION:300
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-PLAYLIST-TYPE:VOD
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00000.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00001.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00002.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00003.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00004.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00005.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00006.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00007.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00008.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00009.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00010.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00011.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00012.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00013.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00014.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00015.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00016.vtt
#EXTINF:199.40000,
V09140_sub_sv/00000/V09140_sub_sv_00017.vtt
#EXT-X-ENDLIST
```

**Comment by spaam** [comment]
interesting :D

**Comment by spaam** [comment]
this fix will be in the next version on sunday

**Comment by filigranen** [comment]
Hmm... Still getting the same results. :-(
```
tomteverkstan] ~/svtviafreesub/svtplay-dl
(01:57) (svt:0) $ git status                                                 On branch master
Your branch is up to date with 'origin/master'.
nothing to commit, working tree clean                                        

[tomteverkstan] ~/svtviafreesub/svtplay-dl
(01:57) (svt:0) $ git log -1                                                 commit cb3612b313975f944cda56bad513eb8f09faebc8 (HEAD -> master, origin/master, origin/HEAD)
Author: Johan Andersson <j@i19.se>
Date:   Fri Mar 5 00:41:49 2021 +0100

    viafree: they have m3u8 subtitles in json

    For some reason they got some bad data in the first file that we need to skip
    fixes: #1332

[tomteverkstan] ~/svtviafreesub/svtplay-dl
(01:57) (svt:0) $ cd ..                                                      

[tomteverkstan] ~/svtviafreesub
(01:57) (svt:0) $ ./svtplay-dl/bin/svtplay-dl --verbose -S --force-subtitle https://www.viafree.se/program/drama/morden-i-midsomer/sasong-16/avsnitt-3
DEBUG [1614905886.3908916] /usr/lib/python3.9/site-packages/svtplay_dl/utils/getmedia.py/get_media: version: 3.0
DEBUG [1614905886.3931584] /usr/lib/python3.9/site-packages/svtplay_dl/service/__init__.py/__init__: service: viaplay
DEBUG [1614905886.3932037] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://www.viafree.se/program/drama/morden-i-midsomer/sasong-16/avsnitt-3'
DEBUG [1614905886.3942235] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): www.viafree.se:443
DEBUG [1614905886.4617314] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://www.viafree.se:443 "GET /program/drama/morden-i-midsomer/sasong-16/avsnitt-3 HTTP/1.1" 200 None
DEBUG [1614905886.4652214] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://viafree.mtg-api.com/identity/viafree/auth/pwd/sessions'
DEBUG [1614905886.4665072] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): viafree.mtg-api.com:443
DEBUG [1614905886.8707843] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://viafree.mtg-api.com:443 "POST /identity/viafree/auth/pwd/sessions HTTP/1.1" 200 None
DEBUG [1614905886.873313] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://viafree.mtg-api.com/stream-links/viafree/web/se/clear-media-guids/V09140/streams'
DEBUG [1614905886.9626505] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://viafree.mtg-api.com:443 "GET /stream-links/viafree/web/se/clear-media-guids/V09140/streams HTTP/1.1" 200 983
DEBUG [1614905886.9633675] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://vod9-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net/e6ca938/V09140-1611600129762.ism/master.m3u8?start_index=1&filter=%28type%3D%3D%22audio%22%26%26%28FourCC%3D%3D%22AACL%22%29%29%7C%7C%28type%3D%3D%22video%22%26%26FourCC%3D%3D%22AVC1%22%29%7C%7C%28type%3D%3D%22textstream%22%26%26%28systemLanguage%3D%3D%22swe%22%29%29&hdnts=st=1614905826~exp=1614992286~acl=*~hmac=13674d98fecf825d1dd0357d4e20f780639e451239da3ba7ff1504db1f4badd7'
DEBUG [1614905886.966181] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): vod9-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net:443
DEBUG [1614905887.0933576] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://vod9-hls-cdn1-vf-cdn-viaplay-tv.akamaized.net:443 "GET /e6ca938/V09140-1611600129762.ism/master.m3u8?start_index=1&filter=%28type%3D%3D%22audio%22%26%26%28FourCC%3D%3D%22AACL%22%29%29%7C%7C%28type%3D%3D%22video%22%26%26FourCC%3D%3D%22AVC1%22%29%7C%7C%28type%3D%3D%22textstream%22%26%26%28systemLanguage%3D%3D%22swe%22%29%29&hdnts=st=1614905826~exp=1614992286~acl=*~hmac=13674d98fecf825d1dd0357d4e20f780639e451239da3ba7ff1504db1f4badd7 HTTP/1.1" 200 None
DEBUG [1614905887.1010206] /usr/lib/python3.9/site-packages/svtplay_dl/utils/http.py/request: HTTP getting 'https://cdn-subtitles-mtgx-tv.akamaized.net/pitcher/V0xxxx/V091xx/V09140/6070025811/V09140_sub_sv.m3u8'
DEBUG [1614905887.1027532] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_new_conn: Starting new HTTPS connection (1): cdn-subtitles-mtgx-tv.akamaized.net:443
DEBUG [1614905887.2577085] /usr/lib/python3.9/site-packages/urllib3/connectionpool.py/_make_request: https://cdn-subtitles-mtgx-tv.akamaized.net:443 "GET /pitcher/V0xxxx/V091xx/V09140/6070025811/V09140_sub_sv.m3u8 HTTP/1.1" 200 1249
INFO [1614905887.2586207] /usr/lib/python3.9/site-packages/svtplay_dl/utils/output.py/output: Outfile: morden.i.midsomer.s16e03.avsnitt.3-685979-viaplay.srt

[tomteverkstan] ~/svtviafreesub
(01:58) (svt:0) $ cat morden.i.midsomer.s16e03.avsnitt.3-685979-viaplay.srt
#EXTM3U
#EXT-X-TARGETDURATION:300
#EXT-X-VERSION:3
#EXT-X-MEDIA-SEQUENCE:0
#EXT-X-PLAYLIST-TYPE:VOD
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00000.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00001.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00002.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00003.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00004.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00005.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00006.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00007.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00008.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00009.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00010.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00011.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00012.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00013.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00014.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00015.vtt
#EXTINF:300.00000,
V09140_sub_sv/00000/V09140_sub_sv_00016.vtt
#EXTINF:199.40000,
V09140_sub_sv/00000/V09140_sub_sv_00017.vtt
#EXT-X-ENDLIST
```

**Comment by filigranen** [comment]
Bah. N/m. Forgot to build a new package after pulling. /facepalm
Thanks for the quick fix!
